staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -77,7 +78,7 @@ func (t *hasTerm) RequiresExactMatch(field string) (value string, found bool) {
 func (t *hasTerm) Transform(fn TransformFunc) (Selector, error) {
 	field, value, err := fn(t.field, t.value)
 	if err != nil {
-		return nil, err
+		return nil, errors.ErrorToBadRequest(err)
```

Comment:
I'm trying to improve the quality of our response codes so that they reflect what is going wrong. Currently we always return a 500 no matter what went wrong. My impression of the errors coming out of the fields and labels packages is that they indicate that the pieces passed in be the client (value or label) are bad and will never work with this version of the API server. In that case 400 is the more appropriate HTTP error.